### PR TITLE
Backports from the thrimbornet branch

### PR DESCRIPTION
**Pull request by dracc**
During work targeting @thrimbor's NIC driver fixes, some oddities were identified and corrected.

- The framebuffer is now allocated
- FTP connections no longer kill themselves - the FTP server kills them instead
- FTP connections now have a decent destructor
- `outputLine` no longer requires SDL in the non-nxdk build

### CODE REVIEW

File: Includes/ftpConnection.cpp [hunk 1]
```diff
@@ -38,6 +38,8 @@
 bool nxIsDriveMounted(char) { return true; }
 #endif
 
+#define FTP_CMD_BUFFER_SIZE (1024)
+#define FTP_DATA_BUFFER_SIZE (64 * 1024)
```
Comment (Teufelchen1) [inline Includes/ftpConnection.cpp:42]: Whats your reason for this exact sizes?

File: Includes/ftpConnection.cpp [hunk 2]
```diff
@@ -78,28 +80,39 @@ void ftpConnection::sendStdString(int fd, std::string const& s, int flags = 0) {
 
 ftpConnection::ftpConnection(int fd, ftpServer* s) :
   _fd(fd), server(s) {
+  buf = (char*)malloc(FTP_CMD_BUFFER_SIZE);
```
Comment (Teufelchen1) [inline Includes/ftpConnection.cpp:83]: Unchecked malloc? Is it safe to assume `buf != NULL`?

File: Includes/ftpConnection.cpp [hunk 3]
```diff
@@ -495,7 +510,7 @@ bool ftpConnection::sendFile(std::string const& fileName) {
     outputLine("File opening failed. LOL.\n");
     return false;
   }
-  int bytesToRead = FTP_BUFFER_SIZE - 3;
+  int bytesToRead = FTP_DATA_BUFFER_SIZE - 3;
```
Comment (Teufelchen1) [inline Includes/ftpConnection.cpp:513]: I don't understand where the -3 comes from 🤔 not part of the PR tho

Comment (dracc) [reply Includes/ftpConnection.cpp:513]: It was added based on a bad assumption on my part. Could safely be removed. Thanks!

File: Includes/ftpConnection.cpp (hunk 2 above)
Comment (dracc) [reply Includes/ftpConnection.cpp:81]: Probably not safe, no. I've added an `assert()`. Thanks!

File: Includes/ftpConnection.cpp (hunk 1 above)
Comment (dracc) [reply Includes/ftpConnection.cpp]: I've added a comment. Hope it added some clarity. :)

File: Includes/ftpConnection.cpp (hunk 2 above)
Comment (Teufelchen1) [reply Includes/ftpConnection.cpp:81]: Not sure if an assert is a good solution. How does nevolutionX behave if it runs out of memory? Does the whole application crash then? Does this break xboxes permanently when it is used as the default dashboard. (ofc, its highly unlikely but still...)

Comment (dracc) [reply Includes/ftpConnection.cpp:81]: You're right (as so often before). Now there's a check in the beginning of `update()` instead.
